event/event.go

```diff
@@ -239,11 +241,59 @@ type ReleaseEventCommon struct {
 	Error string `json:"error,omitempty"`
 }
 
+const (
+	// ReleaseImageSpecType is a type of release spec when there are update.Images
+	ReleaseImageSpecType = "releaseImageSpecType"
+	// ReleaseContainersSpecType is a type of release spec when there are update.Containers
+	ReleaseContainersSpecType = "releaseContainersSpecType"
+)
+
+// ReleaseSpec is a spec for images and containers release
+type ReleaseSpec struct {
+	// Type is ReleaseImageSpecType or ReleaseContainersSpecType
+	// if empty (for previous version), then use ReleaseImageSpecType
+	Type                  string
+	ReleaseImageSpec      *update.ReleaseImageSpec
+	ReleaseContainersSpec *update.ReleaseContainersSpec
+}
+
+// UnmarshalJSON for old version of spec (update.ReleaseImageSpec) where Type is empty
+func (s *ReleaseSpec) UnmarshalJSON(b []byte) error {
+	type T ReleaseSpec
+	t := (*T)(s)
+	if err := json.Unmarshal(b, &t); err != nil {
```

Comment:
`t` is already a pointer `*T`, so you don't need to pass its address.

Reply:
Fixed.